src/main/java/org/maxgamer/quickshop/shop/ShopControlPanel.java

```diff
@@ -0,0 +1,190 @@
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.maxgamer.quickshop.QuickShop;
+import org.maxgamer.quickshop.chat.QuickComponent;
+
+import java.util.Map;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+public class ShopControlPanel {
+    private final Map<Plugin, TreeSet<Entry>> panelRegistry = new ConcurrentHashMap<>();
+    private final QuickShop plugin;
+
+    public ShopControlPanel(QuickShop plugin) {
+        this.plugin = plugin;
+    }
+
+
+    /**
+     * Register the ControlPanel Entry to registry
+     *
+     * @param namespace The plugin instance
+     * @param priority  The entry priority
+     * @param executor  The executor, should returns an QuickComponent or null if no out-put needs
+     */
+    public void register(@NotNull Plugin namespace, int priority, @NotNull Function<Map.Entry<@NotNull CommandSender, @NotNull Shop>, @Nullable QuickComponent> executor) {
+        TreeSet<Entry> entries = this.panelRegistry.get(namespace);
+        if (entries == null) {
+            entries = new TreeSet<>(new Comparator());
+        }
+        Entry entry = new Entry(priority, executor);
+        if (entries.contains(entry)) {
+            return;
+        }
+        entries.add(entry);
+        this.panelRegistry.put(namespace, entries);
+    }
+
+    /**
+     * Unregister a registered executor from registry
+     *
+     * @param namespace The plugin instance
+     * @param priority  The entry priority
+     * @param executor  The executor, should returns an QuickComponent or null if no out-put needs
+     */
+    public void unregister(@NotNull Plugin namespace, int priority, @NotNull Function<Map.Entry<@NotNull CommandSender, @NotNull Shop>, @Nullable QuickComponent> executor) {
+        TreeSet<Entry> entries = this.panelRegistry.get(namespace);
+        if (entries == null) {
+            return;
+        }
+        Entry entry = new Entry(priority, executor);
+        entries.remove(entry);
+    }
+
+    /**
+     * Unregister a registered executor from registry
+     *
+     * @param namespace The plugin instance
+     * @param executor  The executor, should returns an QuickComponent or null if no out-put needs
+     */
+    public void unregister(@NotNull Plugin namespace, @NotNull Function<Map.Entry<@NotNull CommandSender, @NotNull Shop>, @Nullable QuickComponent> executor) {
+        TreeSet<Entry> entries = this.panelRegistry.get(namespace);
+        if (entries == null) {
+            return;
+        }
+        entries.removeIf(entry -> entry.getExecutor().equals(executor));
+    }
+
+    /**
+     * Unregister plugin's all registered executor from registry
+     *
+     * @param namespace The plugin instance
+     */
+    public void unregisterAll(@NotNull Plugin namespace) {
+        this.panelRegistry.remove(namespace);
+    }
+
+
+    public static class Entry {
```

Comment:
Please give a proper name for this class, for example `ControlPanelEntry`

Reply:
It is the inner class it would look like `ShopControlPanel.Entry` so it's okay